instance/tasks.py

```diff
@@ -75,12 +76,15 @@ def spawn_appserver(
                 instance.tags.add(success_tag)
             if mark_active_on_success:
                 # If the AppServer provisioned successfully, make it active.
-                make_appserver_active(
-                    appserver_id,
-                    active=mark_active_on_success,
-                    deactivate_others=deactivate_old_appservers
-                )
-            break
+                appserver = OpenEdXAppServer.objects.get(pk=appserver_id)
+
+                if appserver.server.status == OpenStackServer.Status.Ready:
```

Comment:
This is the wrong level at which to check healthiness. It is *one* place, but we want to cover all possible places where activating an appserver takes place, and this is only one of them.

So we need to go to a lower level, in particular in the `make_appserver_active` function itself. It would be good to also log a note there for when the OpenStack server status is not correct for whatever reason.

Reply:
Also, I think we should import `Server` rather than `OpenStackServer` to get `.Status.Ready`, since it lets it be more generic for the future (i.e. an `AwsServer` would also be covered).